lib/dal/src/func/binding/critical_section.rs

```diff
@@ -0,0 +1,104 @@
+//! This modules contains [`execute_critical_section`].
+
+use veritech_client::{BeforeFunction, ResolverFunctionComponent};
+
+use crate::func::backend::array::FuncBackendArray;
+use crate::func::backend::boolean::FuncBackendBoolean;
+use crate::func::backend::diff::FuncBackendDiff;
+use crate::func::backend::identity::FuncBackendIdentity;
+use crate::func::backend::integer::FuncBackendInteger;
+use crate::func::backend::js_action::FuncBackendJsAction;
+use crate::func::backend::js_attribute::{FuncBackendJsAttribute, FuncBackendJsAttributeArgs};
+use crate::func::backend::js_reconciliation::FuncBackendJsReconciliation;
+use crate::func::backend::js_schema_variant_definition::FuncBackendJsSchemaVariantDefinition;
+use crate::func::backend::json::FuncBackendJson;
+use crate::func::backend::map::FuncBackendMap;
+use crate::func::backend::object::FuncBackendObject;
+use crate::func::backend::string::FuncBackendString;
+use crate::func::backend::validation::FuncBackendValidation;
+use crate::func::backend::{FuncBackend, FuncBackendError, FuncDispatch, FuncDispatchContext};
+use crate::func::binding::{FuncBindingError, FuncBindingResult};
+use crate::{Func, FuncBackendKind};
+
+/// Perform function execution to veritech for a given [`Func`] and [`FuncDispatchContext`].
+///
+/// This function does not persist data to "layerdb". If data persistence for execution is required,
+/// use [`FuncBinding::create_and_execute`](crate::FuncBinding::create_and_execute).
+///
+/// _Warning:_ this function should be used with caution!
+pub(crate) async fn execute_critical_section(
```

Comment:
I went back and forth on splitting this out. Ultimately, the idea of "here's my func, args, and before funcs, let's roll" is a solid bounded context. However, we _really_ do not want multiple ways to execute funcs, so "test execute func" and `FuncBinding::create_and_execute` should be the only two things that call this function. Let's keep an eye on this moving forward.